drivers/clock_control/clock_stm32f2_f4_f7.c

```diff
@@ -59,6 +60,24 @@ void config_pll_sysclock(void)
 				    pllm(STM32_PLL_M_DIVISOR),
 				    STM32_PLL_N_MULTIPLIER,
 				    pllp(STM32_PLL_P_DIVISOR));
+
+#if defined(CONFIG_SOC_SERIES_STM32F7X)
+	/* Assuming we stay on Power Scale default value: Power Scale 1 */
+	if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC > 180000000) {
+		/* Enable the PLL (PLLON) before setting overdrive */
+		LL_RCC_PLL_Enable();
```

Comment:
`LL_RCC_PLL_Enable()` is already part of `set_up_plls()`
Why not moving this to `set_up_plls()` just after `LL_RCC_PLL_Enable`.
Or just move it after `set_up_plls()`

Reply:
This was my thought train:
So I moved the overdrive part out of the common driver and into the f2_f4_f7 driver, which in my opinion is keeping the common clean. Changes made to the common driver should not impact this "side step" only for the F7, so that's why I added an extra `LL_RCC_PLL_Enable()`.  

So if you want this to change, the overdrive part will move back into the common driver and into the funtion `set_up_plls()`. Is this what you want?

Reply:
> which in my opinion is keeping the common clean

That's nice of you.

> Changes made to the common driver should not impact this "side step" only for the F7, so that's why I added an extra `LL_RCC_PLL_Enable()`.

One point is that you're skipping the `while (LL_RCC_PLL_IsReady() != 1U)` step, which I think should be added. So I agree with you that this additional step should be quick, impacts only F7 and has a quite marginal impact (no pm support on F7, so happens only at boot).

Reply:
I see what you mean. This is on purpose, the PLLRDY bit indicates the PLL is locked.
And according to the reference manual this is not needed before changing the overdrive.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/37693428/c615ff21-1215-462b-8d91-6b0b89e2bec3)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/37693428/795cfb33-2a26-4824-b0e7-084595c03052)

Reply:
ok, then can you add a quick comment about it ? (to save time for the next one reading this code) and this will be fine for me.

Reply:
Yeah sure!